src/block/inventory/EnderChestInventory.php

```diff
@@ -85,6 +86,14 @@ protected function internalSetContents(array $items) : void{
 		$this->inventory->setContents($items);
 	}
 
+	public function getViewerCount() : int{
+		$enderChest = $this->getHolder()->getWorld()->getTile($this->getHolder());
+		if(!$enderChest instanceof EnderChest){
+			throw new \UnexpectedValueException('Tile is not an Enderchest');
```

Comment:
I think it's OK to just return 0 in this case. It's possible for the tile to be missing or incorrect due to plugin interference, older worlds using setBlock(), the block being destroyed, etc.

Throwing here will make the server crash if the tile doesn't exist. It's not really a bug in the server, so it shouldn't do that.